projects/vdk-core/tests/functional/ingestion/test_ingest_chain.py

```diff
@@ -9,95 +9,227 @@
-# 2) INGEST_PAYLOAD_PREPROCESS_SEQUENCE=A,B and VDK_INGEST_METHOD_DEFAULT=C
-# 3) INGEST_PAYLOAD_PREPROCESS_SEQUENCE=A,B and VDK_INGEST_METHOD_DEFAULT=C and
-#    job code is D
-# 4) INGEST_PAYLOAD_PREPROCESS_SEQUENCE=""
-# 5) INGEST_PAYLOAD_PREPROCESS_SEQUENCE=A,B - A,B are used for pre-processing,
-#    B is used for ingesting and post-processing
-
 
 log = logging.getLogger(__name__)
 
 
+# INGEST_PAYLOAD_PREPROCESS_SEQUENCE=""
+@mock.patch.dict(
+    os.environ,
+    {"VDK_INGEST_METHOD_DEFAULT": "memory"},
+)
+def test_ingest_no_preprocessing():
+    ingest_plugin = IngestIntoMemoryPlugin()
+    runner = CliEntryBasedTestRunner(ingest_plugin)
+
+    result: Result = runner.invoke(
+        ["run", jobs_path_from_caller_directory("no-ingest-method-job")]
+    )
+
+    cli_assert_equal(0, result)
+
+    expected_object = dict(
+        int_key=42,
+        str_key="example_str",
+        bool_key=True,
+        float_key=1.23,
+        nested=dict(key="value"),
+    )
+    assert ingest_plugin.payloads[0].payload[0] == expected_object
+    assert ingest_plugin.payloads[0].destination_table == "object_table"
+
+    expected_rows_object = {"first": "two", "second": 2}
+    assert ingest_plugin.payloads[1].payload[0] == expected_rows_object
+    assert ingest_plugin.payloads[1].destination_table == "tabular_table"
+
+
+# INGEST_PAYLOAD_PREPROCESS_SEQUENCE=A,B in job code method = C
+@mock.patch.dict(
+    os.environ,
+    {"VDK_INGEST_PAYLOAD_PREPROCESS_SEQUENCE": "convert-to-string," "add-payload-size"},
+)
+def test_chained_ingest_direct_method_passed():
+    pre_ingest_plugin = ConvertPayloadValuesToString()
+    payload_size_plugin = AddPayloadSizeAsColumn()
+    ingest_plugin = IngestIntoMemoryPlugin()
+    runner = CliEntryBasedTestRunner(
+        pre_ingest_plugin, payload_size_plugin, ingest_plugin
+    )
+
+    # Use a sample data job, in which `method` argument is passed to ingestion
+    # method calls.
+    result: Result = runner.invoke(
+        ["run", jobs_path_from_caller_directory("test-ingest-job")]
+    )
+
+    cli_assert_equal(0, result)
+
+    expected_object = dict(
+        int_key=str(42),
+        str_key="example_str",
+        bool_key=str(True),
+        float_key=str(1.23),
+        nested=str(dict(key="value")),
+        payload_size=163,
+    )
+    assert ingest_plugin.payloads[0].payload[0] == expected_object
+    assert ingest_plugin.payloads[0].destination_table == "object_table"
+
+    expected_rows_object = {"first": "two", "payload_size": 80, "second": "2"}
+    assert ingest_plugin.payloads[1].payload[0] == expected_rows_object
+    assert ingest_plugin.payloads[1].destination_table == "tabular_table"
+
+
+# INGEST_PAYLOAD_PREPROCESS_SEQUENCE=A,B no ingest method specified
+@mock.patch.dict(
+    os.environ,
+    {"VDK_INGEST_PAYLOAD_PREPROCESS_SEQUENCE": "convert-to-string," "add-payload-size"},
+)
+def test_chained_ingest_no_method_passed():
```

Comment:
In that case would the preprocess plugins be invoked ? 

I think it's fine either way as long as it's documented and we verification in the test here whichever it is .

Reply:
No, pre-processors will not be invoked. I've added a note about this behaviour. Thanks